### PR TITLE
Cmake Tests: Delete operator check for size_t arg

### PR DESCRIPTION
**Pull request by SAtacker**
- Clang fails to recognize the global delete operator which has 2 arguments the second one being std::size_t
- Reference - https://godbolt.org/z/q1hMrfKjv
- Fixes https://github.com/STEllAR-GROUP/hpx/issues/6246

Signed-off-by: Shreyas Atre <shreyasatre16@gmail.com>

### CODE REVIEW

File: libs/core/type_support/include/hpx/type_support/generator.hpp [hunk 1]
```diff
@@ -324,8 +324,13 @@ namespace hpx {
 
                 dealloc_fn const dealloc = [](void* const p,
                                                std::size_t const s) {
+#if defined(HPX_WITH_CXX14_DELETE_OPERATOR_WITH_SIZE)
                     ::operator delete[](
                         p, s + sizeof(std::size_t) + sizeof(dealloc_fn));
+#else
+                    ::operator delete[](p);
+                    (void) s;
+#endif
```
Comment (hkaiser) [inline libs/core/type_support/include/hpx/type_support/generator.hpp:333]:
```suggestion
                dealloc_fn const dealloc = [](void* const p,
                    [[maybe_unused]] std::size_t const s) {
#if defined(HPX_WITH_CXX14_DELETE_OPERATOR_WITH_SIZE)
                    ::operator delete[](
                        p, s + sizeof(std::size_t) + sizeof(dealloc_fn));
#else
                    ::operator delete[](p);
#endif
```

Comment (SAtacker) [reply libs/core/type_support/include/hpx/type_support/generator.hpp:333]: @hkaiser maybe_unused is since c++17, what do you suggest?

Comment (SAtacker) [reply libs/core/type_support/include/hpx/type_support/generator.hpp:333]: Ohkay, it'll work out because of the previous checks. Thanks!

**Review by hkaiser — COMMENTED** [review]
@SAtacker please fix the clang-format and inspect issues. Otherwise, LGTM! Thanks!

**Comment by hkaiser** [comment]
bors merge

**Pull request merged**